### PR TITLE
Add type stubs to Markdown module

### LINKED ISSUE #5852: Stubs from Markdown are lacking types
The stubs provided for the Python-Markdown module are lacking types for the parameters and returns for almost every method/function defined.

### PR DESCRIPTION
**Pull request by PythonCoderAS**
This is my effort to add actual type stubs.

Fix #5852

### CODE REVIEW

File: stubs/Markdown/markdown/blockprocessors.pyi [hunk 1]
```diff
@@ -53,7 +59,7 @@ class SetextHeaderProcessor(BlockProcessor):
 class HRProcessor(BlockProcessor):
     RE: str = ...
     SEARCH_RE: Pattern[str]
-    match: Any
+    match: Match[str] # TODO: This type won't be compatible in 3.10.
```
Comment (JelleZijlstra) [inline stubs/Markdown/markdown/blockprocessors.pyi:62]: Can you clarify this? If you're referring to `typing.Match` vs. `re.Match`, either will work in stubs.

Comment (PythonCoderAS) [reply stubs/Markdown/markdown/blockprocessors.pyi:62]: I'm pretty sure `match` becomes a keyword in 3.10. Not sure how that affects stubs.

Comment (hauntsaninja) [reply stubs/Markdown/markdown/blockprocessors.pyi:62]: The new PEG parser in 3.9 allows for "soft" keywords, so you can still use match as a variable name or whatever

Comment (PythonCoderAS) [reply stubs/Markdown/markdown/blockprocessors.pyi:62]: Oh, really? That's very interesting to hear. I haven't really upgraded my Python above 3.8 so I haven't looked into all of the big differences after.

Comment (Akuli) [reply stubs/Markdown/markdown/blockprocessors.pyi:62]: A bit off-topic, but the soft keywords lead to weird tricks that you can do (don't try at home, lol):

```python
>>> case = print
>>> match = print
>>> match match:
...     case case:
...             match(match(case(case)))
... 
<built-in function print>
None
None
>>> 
```

**Comment by JelleZijlstra** [comment]
Could you fix the various CI issues and remove the incorrect comment about `match`?

**Comment by PythonCoderAS** [comment]
Sorry, I've been very busy recently, I'll work on it tomorrow.

**Comment by JelleZijlstra** [comment]
Hi @PythonCoderAS, any updates?

**Comment by PythonCoderAS** [comment]
Sorry, I've lost interest in this PR, if anyone would like to pick it up they are welcome to.

**Pull request closed without merging**